### PR TITLE
EngineBuilder API: addDnsQueryTimeoutSeconds

### PR DESCRIPTION
**Pull request by junr03**
Description: API for addDnsQueryTimeoutSeconds to configure behavior introduced in https://github.com/envoyproxy/envoy-mobile/pull/1580
Risk Level: low - optional builder API
Testing: updated suite.
Docs Changes: added

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

File: test/swift/EngineBuilderTests.swift [hunk 1]
```diff
@@ -19,6 +19,7 @@ fixture_template:
 
 private struct TestFilter: Filter {}
 
+// swiftlint:disable:next type_body_length
```
Comment (junr03) [inline test/swift/EngineBuilderTests.swift:22]: not sure if this is the kosher way of doing this @rebello95

Comment (rebello95) [reply test/swift/EngineBuilderTests.swift:22]: this is fine, or honestly we could just remove this rule from `.swiftlint.yml`

Comment (rebello95) [reply test/swift/EngineBuilderTests.swift:22]: the "fix" would be to split this file up

**Comment by junr03** [comment]
@rebello95 @buildbreaker ready for review

File: library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java [hunk 2]
```diff
@@ -38,6 +39,7 @@
    * @param dnsRefreshSeconds            rate in seconds to refresh DNS.
    * @param dnsFailureRefreshSecondsBase base rate in seconds to refresh DNS on failure.
    * @param dnsFailureRefreshSecondsMax  max rate in seconds to refresh DNS on failure.
+   * @param dnsQueryTimeoutSeconds            rate in seconds to timeout DNS queries.
```
Comment (rebello95) [inline library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java:42]: Nit: indentation

**Review by rebello95 — DISMISSED** [review]

**Comment by junr03** [comment]
@buildbreaker can I get a review to re-approve @rebello95's dismissed review?

**Review by buildbreaker — APPROVED** [review]
I've never been a fan of file length linters (we do this at Lyft's iOS code base actually)

Maybe we can add a task to remove that lint check?

Reference comment: https://github.com/envoyproxy/envoy-mobile/pull/1583/files#r667261030

**Pull request merged**